### ISSUE TITLE
Crash on starting a video call

### ISSUE THREAD

**Issue by EionRobb** [EionRobb/purple-hangouts-attempt2#97]
**[Original report](https://bitbucket.org/EionRobb/purple-hangouts/issue/97) by marcusmueller (Bitbucket: [marcusmueller](https://bitbucket.org/marcusmueller), GitHub: [marcusmueller](https://github.com/marcusmueller)).**

----------------------------------------

Backtrace; commit https://bitbucket.org/marcusmueller/purple-hangouts/commits/9de174d700e9732140922d24dde9b96f9c86ea94 (referring to an out-of-mainline commit, because I had to fix a NULL deref to get to this NULL deref):

```
#!gdb

Program received signal SIGSEGV, Segmentation fault.
hangout_participant_add_cb (ha=0x555556381820, response=0x555556b6a360, user_data=0x55555674e150) at hangouts_media.c:719
719		hangouts_media->hangout_cookie = g_strdup(response->sync_metadata->hangout_cookie->cookie);
(gdb) p hangouts_media
$1 = (HangoutsMedia *) 0x55555674e150
(gdb) p hangouts_media->hangout_cookie
$2 = (gchar *) 0x0
(gdb) p response
$3 = (HangoutParticipantAddResponse *) 0x555556b6a360
(gdb) p response->sync_metadata 
$4 = (SyncMetadata *) 0x0

```

In other words, we *must* check whether `sync_metadata` exists. Not quite sure we should only be ensuring this here (i.e. on access), or where that response object is created; hence no Pull Request prior to decision :)
